### PR TITLE
Test commission change.

### LINKED ISSUE #412: Watch commission of enrolled validators, or confirm that it cannot be changed
Continuation of https://github.com/ChorusOne/solido/pull/410#discussion_r703042549.

We should confirm that only the authorized withdrawer can change the commission percentage of a vote account, ideally by adding a test in `tests/solana_assumptions.rs`.

If it turns out that it is possible for somebody other than the withdraw authority to change the commission percentage, then we should add monitoring so we can detect if any of the validator commission percentages is not the expected 100%. I think the best way to expose that would be to add a metric for it, and then we can set Prometheus alerts on it. But before we go there, let’s verify whether it is necessary.

### PR DESCRIPTION
**Pull request by enriquefynn**
Test that only the withdraw authority can change the commission.
Closes #412.

### CODE REVIEW

File: program/tests/tests/solana_assumptions.rs [hunk 1]
```diff
@@ -258,3 +264,122 @@ async fn test_stake_accounts() {
...
+) -> solana_sdk::transport::Result<()> {
+    send_transaction(
+        &mut context.context,
+        &mut context.nonce,
+        &[vote_instruction::update_commission(
+            &vote_account,
+            &withdraw_authority.pubkey(),
+            new_commission,
+        )],
+        vec![&withdraw_authority],
```
Comment (ruuda) [inline program/tests/tests/solana_assumptions.rs:282]: In this case we pass the withdraw authority both to the `update_commission` instruction, and we use it as the signer. But what if we pass the right withdraw authority, but sign with something else, like the private key of the vote account? Is changing the commission still rejected then?

**Review by ruuda — COMMENTED** [review]
Thanks!

File: program/tests/tests/solana_assumptions.rs (hunk 1 above)
Comment (enriquefynn) [reply program/tests/tests/solana_assumptions.rs:282]: I hope so. Added some changes 👍

Comment (enriquefynn) [reply program/tests/tests/solana_assumptions.rs]: The problem is that signing panics with `KeypairPubkeyMismatch`. I don't think we need to test this assumption :P

Comment (ruuda) [reply program/tests/tests/solana_assumptions.rs]: Hmm, ok, good to know.

**Review by ruuda — APPROVED** [review]
LGTM!

**Pull request merged**